units/provisioning_create_host.go

```diff
@@ -116,13 +116,17 @@ func (j *createHostJob) Run(ctx context.Context) {
 		}
 		if j.host == nil {
 			//host intent document has been removed by another evergreen process
+			notFoundError := errors.New("could not find host")
 			grip.Warning(message.Fields{
 				"host_id": j.HostID,
 				"task_id": j.TaskID,
 				"attempt": j.RetryInfo().CurrentAttempt,
 				"job":     j.ID(),
-				"message": "could not find host",
+				"message": notFoundError.Error(),
 			})
+			if err := task.AddHostCreateDetails(j.host.StartedBy, j.host.Id, j.host.SpawnOptions.TaskExecutionNumber, notFoundError); err != nil {
```

Comment:
iirc `host.list` relies on the number of errors in host create details in the task corresponding to the number of failed hosts the task tried to spawn. If so:
If it's in fact the case that the host document doesn't exist here is it definitely going to be because of an error spawning it? And is that error definitely not going to have added an entry in host create details?

Reply:
i added an extra condition to the current condition of maxing out the attempts